brainscore_vision/model_helpers/activations/core.py

```diff
@@ -209,28 +336,75 @@ def _package_layer(self, layer_activations, layer, stimuli_paths):
             self._logger.debug(f"Unknown layer activations shape {layer_activations.shape}, not inferring channels")
 
         # build assembly
-        coords = {'stimulus_path': stimuli_paths,
-                  'neuroid_num': ('neuroid', list(range(activations.shape[1]))),
-                  'model': ('neuroid', [self.identifier] * activations.shape[1]),
-                  'layer': ('neuroid', [layer] * activations.shape[1]),
-                  }
+        if require_variance:
+            coords = self.build_microsaccade_coords(activations, layer, stimuli_paths)
+        else:
+            coords = {'stimulus_path': ('presentation', stimuli_paths),
+                      'stimulus_path2': ('presentation', stimuli_paths),  # to avoid DataAssembly dim collapse
+                      'neuroid_num': ('neuroid', list(range(activations.shape[1]))),
+                      'model': ('neuroid', [self.identifier] * activations.shape[1]),
+                      'layer': ('neuroid', [layer] * activations.shape[1]),
+                      }
```

Comment:
```suggestion
        coords = {'stimulus_path': ('presentation', stimuli_paths),
                  'stimulus_path2': ('presentation', stimuli_paths),  # to avoid DataAssembly dim collapse
                  'neuroid_num': ('neuroid', list(range(activations.shape[1]))),
                  'model': ('neuroid', [self.identifier] * activations.shape[1]),
                  'layer': ('neuroid', [layer] * activations.shape[1]),
              }
          if require_variance:
              microsaccade_coords = self.build_microsaccade_coords(activations, layer, stimuli_paths)
              coords = {**coords, **microsaccade_coords}
            
```